ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx

```diff
@@ -113,13 +140,29 @@ export default function AssetList({
                 {token.type === AssetType.native ? (
                   <TokenListItem
                     title={nativeCurrency}
-                    primary={
-                      primaryCurrencyProperties.value ??
-                      secondaryCurrencyProperties.value
+                    primary={getPrimaryValue({
+                      useNativeCurrencyAsPrimaryCurrency,
+                      primaryCurrencyDisplay,
+                      showFiat,
+                      secondaryCurrencyDisplay,
+                      isOriginalNativeSymbol,
+                    })}
+                    tokenSymbol={
+                      useNativeCurrencyAsPrimaryCurrency
+                        ? primaryCurrencyProperties.suffix
+                        : secondaryCurrencyProperties.suffix
                     }
-                    tokenSymbol={primaryCurrencyProperties.suffix}
-                    secondary={secondaryCurrencyDisplay}
+                    secondary={getSecondaryValue({
+                      useNativeCurrencyAsPrimaryCurrency,
+                      primaryCurrencyDisplay,
+                      showFiat,
+                      secondaryCurrencyDisplay,
+                      isOriginalNativeSymbol,
+                    })}
                     tokenImage={token.image}
+                    isNativeCurrency
+                    isOriginalTokenSymbol={isOriginalNativeSymbol}
+                    hidePercentage
```

Comment:
Can we avoid this extra logic and the util functions for now? This is overscoped for this PR

Reply:
This process will become simpler once the `useNativeCurrencyAsPrimaryCurrency` option is eliminated. However, currently, we must manage the transition between fiat and crypto on the settings. To address this, I've developed a solution in a utility file, as nested ternary expressions are prohibited. This approach also facilitates the writing of unit tests too.

Reply:
https://github.com/MetaMask/metamask-extension/pull/25601#discussion_r1662706706 applies here as well – this is out of scope and too high risk for this PR